commit dd3961c5c96e17e36ffdf6d162187911ba020d4f
Author: Koen De Vleeschauwer <kdv@kdvelectronics.eu>
Date:   Sun Mar 5 09:04:46 2023 +0100

    arduino-ide

diff --git a/Dockerfile b/Dockerfile
index 64ed71f..d6bbccd 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,12 +1,12 @@
 # hadolint ignore=DL3007
-FROM myoung34/github-runner-base:latest
+FROM myoung34/github-runner-base:ubuntu-bionic
 LABEL maintainer="myoung34@my.apsu.edu"
 
 ENV AGENT_TOOLSDIRECTORY=/opt/hostedtoolcache
 RUN mkdir -p /opt/hostedtoolcache
 
 ARG GH_RUNNER_VERSION="2.302.1"
-ARG TARGETPLATFORM
+ARG TARGETPLATFORM="linux/arm64"
 
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
@@ -21,5 +21,11 @@ RUN chmod +x /actions-runner/install_actions.sh \
 COPY token.sh entrypoint.sh app_token.sh /
 RUN chmod +x /token.sh /entrypoint.sh /app_token.sh
 
+# install arduino ide build dependencies
+RUN apt-get install -y libxkbfile-dev libsecret-1-dev
+
+# install nvm
+RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.1/install.sh | bash
+
 ENTRYPOINT ["/entrypoint.sh"]
 CMD ["./bin/Runner.Listener", "run", "--startuptype", "service"]
diff --git a/entrypoint.sh b/entrypoint.sh
index 5196b7d..193da43 100644
--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -166,6 +166,10 @@ if [[ ${_DISABLE_AUTOMATIC_DEREGISTRATION} == "false" ]]; then
   trap deregister_runner SIGINT SIGQUIT SIGTERM INT TERM QUIT
 fi
 
+. $HOME/.nvm/nvm.sh
+nvm install 14.0.0
+npm install --global yarn
+
 # Container's command (CMD) execution as runner user
 
 
